### PR TITLE
COMMUNITY-79 | Deprecate GH Actions in favour of Jenkins Community

### PR DESCRIPTION
**Pull request by ManuelJNunez**
The following tasks were carried out:
- The "Build & Test" pipeline was refactored and now includes a new dynamic parallel stage that will get the Node versions from the pipeline name or from the config file.
- The "Build Doc" pipeline now have a new stage to publish the generated doc when a new tag with format `/v\d+\.\d+\.\d+-doc/` is pushed.

### CODE REVIEW

File: resources/jenkins/build_and_test.groovy [hunk 1]
```diff
@@ -50,82 +119,25 @@ pipeline {
...
-                stages {
-                    stage('Checkout repo') {
-                        steps {
-                            echo "[INFO] Building from ${pwd()}..."
-
-                            checkout scm
-                        }
+                    // If the node versions was not predefined in the job name, read them from the config file.
+                    if(!nodeVersions) {
+                        ciConfig = readYaml(file: "ci_config.yaml")
```
Comment (Alxe) [inline resources/jenkins/build_and_test.groovy:129]:
```suggestion
                        def ciConfig = readYaml(file: "ci_config.yaml")
```

File: resources/jenkins/build_and_test.groovy [hunk 2]
```diff
@@ -50,82 +119,25 @@ pipeline {
...
-                    }
-                }
-                axes {
-                    axis {
-                        name 'NODE_VERSION'
-                        values '17', '18', '20', 'lts', 'latest'
-                    }
-                }
+            steps {
+                script {
```
Comment (Alxe) [inline resources/jenkins/build_and_test.groovy:123]:
```suggestion
```

File: resources/jenkins/build_doc.groovy [hunk 3]
```diff
@@ -12,7 +12,10 @@
 
 pipeline {
     agent {
-        node {
+        dockerfile {
+            additionalBuildArgs  '--build-arg NODE_VERSION=18'
```
Comment (Alxe) [inline resources/jenkins/build_doc.groovy:16]:
```suggestion
            additionalBuildArgs  '--build-arg NODE_VERSION=20'
```

Is there any reason why we're not doing this?

**Review by Alxe — CHANGES_REQUESTED** [review]

File: resources/jenkins/build_and_test.groovy (hunk 2 above)
Comment (ManuelJNunez) [reply resources/jenkins/build_and_test.groovy:123]: No, that's it needed.

File: resources/jenkins/build_and_test.groovy (hunk 1 above)
Comment (ManuelJNunez) [reply resources/jenkins/build_and_test.groovy:129]: look the previous line:

```
def ciConfig = null
```

Comment (ManuelJNunez) [reply resources/jenkins/build_and_test.groovy]: I moved the scope of that variable.

File: resources/jenkins/build_and_test.groovy (hunk 2 above)
Comment (Alxe) [reply resources/jenkins/build_and_test.groovy:123]: I don't know why this comment is there. I reviewed this from Github's VS Code, which might have created the odd comment.

Comment (Alxe) [reply resources/jenkins/build_and_test.groovy:123]: I think this might have been the change to remove `def ciConfig = null`, but I'm not sure.

**Review by Alxe — APPROVED** [review]

**Pull request merged**